packages/typeorm-legacy/src/utils.ts

```diff
@@ -74,3 +74,35 @@ export function parseConnectionConfig(
     return { url: configOrString } as any
   }
 }
+
+function entitiesChanged(
+  prevEntities: any[] | undefined,
+  newEntities: any[]
+): boolean {
+  if (prevEntities?.length !== newEntities?.length) return true
+
+  for (let i = 0; i < prevEntities?.length; i++) {
+    if (prevEntities[i] !== newEntities[i]) return true
+  }
+
+  return false
+}
+
+export async function updateConnectionEntities(
+  connection: Connection,
+  entities: any[]
+) {
+  if (!entitiesChanged(connection.options.entities, entities)) return
+
+  // @ts-expect-error
+  connection.options.entities = entities
+
+  console.warn("[next-auth] ADAPTER_TYPEORM_REBUILDING_METADATA")
+  // @ts-expect-error
+  connection.buildMetadatas()
+
+  if (connection.options.synchronize) {
+    console.warn("[next-auth] ADAPTER_TYPEORM_UPDATING_ENTITIES")
+    await connection.synchronize()
```

Comment:
I thought we wanted to do this regardless of user config, if `process.env.NODE_ENV !== "production"`. I was actually fine by it. Also, one `console.warn` is probably enough.

Reply:
This would effectively disable the `synchronize` option, I feel like that could be bad? Removes the option to not `synchronize` if you need to test something.

That's why I put it here https://github.com/nextauthjs/adapters/blob/faa2c76f28b99b6f5baecac5f95dde56eb9a7af7/packages/typeorm-legacy/src/index.ts#L40 since `updateConnectionEntities` is only useful when not in production.

Yeah, one warning is enough, I agree.

Reply:
OK, I see. What if we did
`if (connection.options.synchronize !== false)`?

So basically sync by default? The user could still disable in not prod if needed?